src/libs/actions/TeachersUnite.js

```diff
@@ -0,0 +1,172 @@
+import _ from 'underscore';
+import Onyx from 'react-native-onyx';
+import lodashGet from 'lodash/get';
+import ONYXKEYS from '../../ONYXKEYS';
+import * as API from '../API';
+import Navigation from '../Navigation/Navigation';
+import CONST from '../../CONST';
+import * as ReportUtils from '../ReportUtils';
+
+let sessionEmail = '';
+let sessionAccountID = 0;
+Onyx.connect({
+    key: ONYXKEYS.SESSION,
+    callback: (val) => {
+        sessionEmail = lodashGet(val, 'email', '');
+        sessionAccountID = lodashGet(val, 'accountID', 0);
+    },
+});
+
+let allPersonalDetails;
+Onyx.connect({
+    key: ONYXKEYS.PERSONAL_DETAILS_LIST,
+    callback: (val) => (allPersonalDetails = val),
+});
+
+/**
+ * @param {String} reportID
+ * @param {String} values.firstName
+ * @param {String} values.phoneOrEmail
+ * @param {String} [values.lastName]
+ */
+// Will be removed once we start using & remove this comment as well :)
+// eslint-disable-next-line no-unused-vars
+function referTeachersUniteVolunteer(reportID, firstName, phoneOrEmail, lastName = '') {
+    // const optimisticData = [
+    //     {
+    //         onyxMethod: Onyx.METHOD.MERGE,
+    //         key: `${ONYXKEYS.COLLECTION.REPORT}${reportID}`,
+    //         value: {
+    //             firstName,
+    //             lastName,
+    //             phoneOrEmail,
+    //         },
+    //     },
+    // ];
+    // API.write(
+    //     'ReferTeachersUniteVolunteer',
+    //     {
+    //         reportID,
+    //         firstName,
+    //         lastName,
+    //         phoneOrEmail,
+    //     },
+    //     {optimisticData},
+    // );
+    Navigation.dismissModal(CONST.TEACHER_UNITE.PUBLIC_ROOM_ID);
+}
+
+/**
+ * Optimistically creates a policyExpenseChat for schoolPrincipal and pass data to AddSchoolPrincipal
+ * @param {String} firstName
+ * @param {String} email
+ * @param {String} [lastName]
+ */
+// Will be removed once we start using & remove this comment as well :)
+// eslint-disable-next-line no-unused-vars
+function createExpenseChatSchoolPrincipal(firstName, email, lastName = '') {
+    const policyName = 'TeacherUniteSchoolPrincipal';
+    const policyID = CONST.TEACHER_UNITE.SCHOOL_PRINCIPAL_POLICY_ID;
+
+    const {expenseChatReportID, expenseChatData, expenseReportActionData, expenseCreatedReportActionID} = ReportUtils.buildOptimisticExpenseChatForSchoolPrincipal(policyID, policyName);
+
+    API.write(
+        'AddSchoolPrincipal',
+        {
+            firstName,
+            lastName,
+            email,
+            expenseCreatedReportActionID,
```

Comment:
Seems this is not needed since we'll be passing `policyExpenseCreatedReportActionID`